TestFoundation/TestNSTimeZone.swift

```diff
@@ -117,6 +117,20 @@ class TestNSTimeZone: XCTestCase {
         let known = NSTimeZone.knownTimeZoneNames
         XCTAssertNotEqual([], known, "known time zone names not expected to be empty")
     }
+    
+    func test_localizedName(){
+        let initialTimeZone = NSTimeZone.default
+        NSTimeZone.default = TimeZone(identifier: "America/New_York")!
+        let defaultTimezone = NSTimeZone.default
+        let localeL = Locale(identifier: "en_US")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.standard, locale: localeL), "Eastern Standard Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortStandard, locale: localeL), "EST")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.generic, locale: localeL), "Eastern Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.daylightSaving, locale: localeL), "Eastern Daylight Time")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortDaylightSaving, locale: localeL), "EDT")
+        XCTAssertEqual(defaultTimezone.localizedName(for: NSTimeZone.NameStyle.shortGeneric, locale: localeL), "ET")
+        NSTimeZone.default = initialTimeZone //reset the Timezone
```

Comment:
Additionally:
* Can we change `localeL` to just `locale`?
* Can we drop `NSTimeZone.NameStyle` so these cases just read `defaultTimeZone.localizedName(for: .standard, locale: locale), ...`?